### PR TITLE
Project structure server

### PR DESCRIPTION
**Pull request by mohamadhase**
1) Update Project structure by adding the files needed to run the back-end server using FAST-API 
2) change the name of src -> source to avoid the conflict happened with the srd module in the python standard modules
3) update the project_structure.txt file with the new files 
4) add utils.py file to store all the helper functions we need in the project 
5) add constants.py to store all the constants values that will be shared in the project
6) add run.py file to be the main file to run the back-end server
7) create Road module and add __init__.py file with all the logic to init the server for the backend

### CODE REVIEW

**Comment by mohamadhase** [comment]
in the last commit i just add git ignore file

File: Road/__init__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+app = FastAPI(
+    title="Road Status API",
+    description="",
+    version=0.1
+)
+logger = logging.getLogger(__name__)
+logging.basicConfig(filename='Road/logs/RouadStatus.log',level=logging.DEBUG)
+
+
+from Road.endpoints import endpoints
```
Comment (mawada-sweis) [inline Road/__init__.py:13]: Importing modules should be at the top of the file, right?

File: Road/__init__.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from fastapi import FastAPI,Request
+from http import HTTPStatus
```
Comment (mawada-sweis) [inline Road/__init__.py:2]: Why did you import `Request` and `HTTPStatus` while you didn't use them?

File: project_structure.txt [hunk 3]
```diff
@@ -1,31 +1,54 @@
...
+    |
+    ├── logs                <- folder conaining all logs files generated from the project
+    │   └── RouadStatus.log       
+    ├── models              <- Trained and serialized models, model predictions, or model summaries
+    |
+    ├── notebooks           <- Jupyter notebooks.
+    |
+    ├── references          <- Data dictionaries, manuals, and all other explanatory materials.
+    |
+    └── src                 <- Source code for use in this project.
```
Comment (mawada-sweis) [inline project_structure.txt:24]: you change the `src` folder name to `source`

File: project_structure.txt [hunk 4]
```diff
@@ -1,31 +1,54 @@
...
+        ├── __init__.py     <- Makes src a Python module (sub module inside Road Module)
+        ├── constants.py    <- file for constant values in the project
+        ├── utils.py        <- file for functions used in the project
```
Comment (mawada-sweis) [inline project_structure.txt:27]: Why do you modify the folder for a single file?

File: project_structure.txt [hunk 5]
```diff
@@ -1,31 +1,54 @@
...
+    ├── notebooks           <- Jupyter notebooks.
+    |
+    ├── references          <- Data dictionaries, manuals, and all other explanatory materials.
+    |
+    └── src                 <- Source code for use in this project.
+        ├── __init__.py     <- Makes src a Python module (sub module inside Road Module)
+        ├── constants.py    <- file for constant values in the project
+        ├── utils.py        <- file for functions used in the project 
+        ├── data            <- Scripts to download or generate data
+        ├── make_dataset.py
```
Comment (mawada-sweis) [inline project_structure.txt:29]: I think you wanted to place this file in the data folder

**Review by mawada-sweis — COMMENTED** [review]
**Overall review:**
- The requirements file is empty, please freeze the necessary libraries.
- Please add comments to explain the code.
- You don't appear to be following the style instructions, please do so.

**Question to answer:**
- Why the `Road-Status` folder contains README file?
-  Why do we have 2 main folder: `Road` and `Road-Status`?
- If possible, including an image describing how these folders/files are related...

File: Road/__init__.py (hunk 2 above)
Comment (mohamadhase) [reply Road/__init__.py:2]: my bad, i wrote some code that uses them then move the code to another file without delete them , i will create a commit to fix it

File: project_structure.txt (hunk 3 above)
Comment (mohamadhase) [reply project_structure.txt:24]: خلاص يبشه 😂 حاضر بغيرها

File: project_structure.txt (hunk 5 above)
Comment (mohamadhase) [reply project_structure.txt:29]: i will check what you mean by make_dataset.py in the other branch before i change and update it if its wrong

File: project_structure.txt (hunk 4 above)
Comment (mohamadhase) [reply project_structure.txt:27]: didn't understand what you mean here  which folder become single file ?

Comment (mawada-sweis) [reply project_structure.txt:27]: Following discussion, this file for generic project functionalities was created.

File: project_structure.txt [hunk 6]
```diff
@@ -1,31 +1,47 @@
...
+    │   ├── interim         <- Intermediate data that has been transformed.
+    │   ├── processed       <- The final, canonical data sets for modeling.
+    │   └── raw             <- The original, immutable data dump.
+    |
+    ├── endpoints
+    │   └── endpoints       <- script containing all endpoints in the server
+    |
+    ├── logs                <- folder conaining all logs files generated from the project
+    │   └── RouadStatus.log       
+    ├── models              <- Trained and serialized models, model predictions, or model summaries
```
Comment (zubaidasader) [inline project_structure.txt:18]: I recommend adding folders for pickled models and models notebooks inside the models folder

**Review by zubaidasader — APPROVED** [review]

File: project_structure.txt (hunk 6 above)
Comment (mohamadhase) [reply project_structure.txt:18]: these folders could be added when we go through the project if we need them , i dont think we need to add them from now and don't use them later  , but if the others agree with the idea there's no problem with adding them

Comment (mawada-sweis) [reply project_structure.txt:18]: I agree with Zubaida, for more organizing...

**Review by Rama-Has — APPROVED** [review]

**Pull request merged**